sematic/plugins/abstract_kuberay_wrapper.py

```diff
@@ -104,10 +141,30 @@ class RayClusterConfig:
     head_node: RayNodeConfig
     scaling_groups: List[ScalingGroup] = field(default_factory=list)
     ray_version: str = field(default_factory=_get_ray_version)
+    autoscaler_config: Optional[AutoscalerConfig] = None
+
+    def __post_init__(self) -> None:
+        if self.requires_autoscale() and self.autoscaler_config is None:
+            raise ValueError(
+                "Your RayClusterConfig would require autoscaling, but no "
+                "AutoScalerConfig is provided. Note that the autoscaler "
+                "will execute in the same Kubernetes pod as the Ray head, "
+                "so you must have a Kubernetes node available which can "
+                "accomodate the resources for the head PLUS the autoscaler."
```

Comment:
How about shortening this message and storing the full story in `errors.md` and adding a link in the exception message?

Reply:
done